.github/workflows/pr.yaml

```diff
@@ -187,3 +187,138 @@ jobs:
+            if echo $TOOL_GROUP | grep -qf .tt_biocontainer_skip; then
+                PLANEMO_OPTIONS=""
+            else
+                PLANEMO_OPTIONS="--biocontainers --no_dependency_resolution --no_conda_auto_init"
+            fi
+            json=$(mktemp -u -p json_output --suff .json)
+            PIP_QUIET=1 planemo test --database_connection postgresql://postgres:postgres@localhost:5432/galaxy $PLANEMO_OPTIONS --galaxy_source $GALAXY_REPO --galaxy_branch $GALAXY_RELEASE --galaxy_python_version ${{ matrix.python-version }} --test_output_json "$json" $TOOL_GROUP || true
+            docker system prune --all --force --volumes || true
+        done < changed_tools_chunk.list
+        if [ ! -s changed_tools_chunk.list ]; then
+            echo '{"tests":[]}' > "$(mktemp -u -p json_output --suff .json)"
+        fi
+    - name: Merge tool_test_output.json files
+      run: planemo merge_test_reports json_output/*.json tool_test_output.json
+    - name: Create tool_test_output.html
+      run: planemo test_reports tool_test_output.json --test_output tool_test_output.html
+    - name: Copy artifacts into place
+      run: |
+        mkdir upload
+        mv tool_test_output.json tool_test_output.html upload/
+    - uses: actions/upload-artifact@v2
+      with:
+        name: 'Tool test output ${{ matrix.chunk  }}'
+        path: upload
+
+  combine_outputs:
+    name: Combine chunked test results
+    needs: [setup, test]
+    strategy:
+      matrix:
+        python-version: [3.8]
+    # This job runs on Linux
+    runs-on: self-hosted
+    steps:
+    - uses: actions/download-artifact@v2
+      with:
+        path: artifacts
+    - uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Cache .cache/pip
+      uses: actions/cache@v2
+      id: cache-pip
+      with:
+        path: ~/.cache/pip
+        key: pip_cache_py_${{ matrix.python-version }}_gxy_${{ needs.setup.outputs.galaxy_head_sha }}
+    - name: Install Planemo
+      run: pip install planemo
+    - name: Install jq
+      run: sudo apt-get install jq
+    - name: Combine outputs
+      run: find artifacts/ -name tool_test_output.json -exec sh -c 'planemo merge_test_reports "$@" tool_test_output.json' sh {} +
+    - name: Create tool_test_output.html
+      run: planemo test_reports tool_test_output.json --test_output tool_test_output.html
+    - name: Copy artifacts into place
+      run: |
+        mkdir upload
+        mv tool_test_output.json tool_test_output.html upload/
+    - uses: actions/upload-artifact@v2
+      with:
+        name: 'All tool test results'
+        path: upload
+    - name: Check status of combined outputs
+      run: |
+        if jq '.["tests"][]["data"]["status"]' upload/tool_test_output.json | grep -v "success"; then
+            echo "Unsuccessful tests found, inspect the 'All tool test results' artifact for details."
+            exit 1
+        fi
+
+#    repo=$(basename "$DIR")
```

Comment:
Maybe tidy up these test lines?

Reply:
ahh, that is Andrey's logic for getting stuff from the FTP, that I had to rescue from some lost branches... we should actually add that.